pagopa/io-app
# fix: [IABT-1386] Fix the Android camera permissions request

## Short description
This PR is going to add the OS permissions request in the QR Code Scanner screen after the rationale showed up on Android.

https://user-images.githubusercontent.com/5963683/183673717-95943319-1c9b-4698-878b-0334dd8dedbc.mov

## List of changes proposed in this pull request
- Moved the rationale logic inside the `BarcodeCamera` component

## How to test
On Android, remove all the permissions from the emulator using `adb shell pm reset-permissions` and navigate to the QR Code Scanner screen. An alert should show up and after pressing the button another alert with the relative permissions request should appear. Giving the correct permissions should make the camera available in the screen.
